code/modules/arousal/arousal.dm

```diff
@@ -68,6 +68,10 @@
 		R.reaction(turfing ? target : target.loc, TOUCH, 1, 0)
 	if(!turfing)
 		R.trans_to(target, R.total_volume * (spill ? G.fluid_transfer_factor : 1))
+
+	overlay_fullscreen("flash", /obj/screen/fullscreen/flash) // To be more true to the hentai
+	clear_fullscreen("flash", 20)
```

Comment:
can't you just use flash_act?

Reply:
actually nevermind that takes into account eye protection

Reply:
It does fadeout